server/manager/src/main/java/org/apache/accumulo/manager/tableOps/delete/DeleteTable.java

```diff
@@ -41,6 +42,19 @@ public DeleteTable(NamespaceId namespaceId, TableId tableId) {
 
   @Override
   public long isReady(long tid, Manager env) throws Exception {
+
+    // Before attempting to delete the table, cancel any running user
+    // compactions.
+    if (Utils.reserveNamespace(env, namespaceId, tid, false, true, TableOperation.COMPACT_CANCEL)
+        + Utils.reserveTable(env, tableId, tid, false, true, TableOperation.COMPACT_CANCEL) == 0) {
+      try {
+        CancelCompactions.mutateZooKeeper(tid, tableId, env);
+      } finally {
+        Utils.unreserveTable(env, tableId, tid, false);
+        Utils.unreserveNamespace(env, namespaceId, tid, false);
```

Comment:
It's too bad we can't hold the read lock and only release it after we get the write lock, blocking the write lock from being taken by anybody else. Delete should have priority.